pkgs/servers/mastodon/default.nix

```
@@ -0,0 +1,84 @@
+{ nodejs-slim, yarn2nix-moretea, fetchFromGitHub, bundlerEnv,
+  stdenv, yarn, lib, callPackage, imagemagick, ffmpeg, file, ... }:
+
+let
+  version = import ./version.nix;
+  src = callPackage ./source-patched.nix {};
+
+  mastodon-gems = bundlerEnv {
+    name = "mastodon-gems";
+    inherit version;
+    gemdir = src;
+    gemset = ./gemset.nix;
+  };
+
+  mastodon-js-modules = yarn2nix-moretea.mkYarnPackage {
+    name = "mastodon-modules";
+    yarnNix = ./yarn.nix;
+    packageJSON = ./package.json;
+    inherit src;
+    inherit version;
+  };
+
+  mastodon-assets = stdenv.mkDerivation {
+    pname = "mastodon-assets";
+    inherit src version;
+
+    buildInputs = [
+      mastodon-gems nodejs-slim yarn
+    ];
+
+    buildPhase = ''
+      cp -r ${mastodon-js-modules}/libexec/*/mastodon/node_modules "node_modules"
+      chmod -R u+w node_modules
+      rake assets:precompile
+    '';
+
+    installPhase = ''
+      mkdir -p $out/public
+      cp -r public/assets $out/public
+      cp -r public/packs $out/public
+    '';
+  };
+
+in stdenv.mkDerivation {
+  pname = "mastodon";
+  inherit src version;
+
+  passthru.updateScript = callPackage ./update.nix {};
+
+  buildPhase = ''
+    ln -s ${mastodon-js-modules}/libexec/*/mastodon/node_modules node_modules
+    ln -s ${mastodon-assets}/public/assets public/assets
+    ln -s ${mastodon-assets}/public/packs public/packs
+
+    for b in $(ls bin/)
+    do
+      if [ -f bin/$b ]; then
+        sed -i 's#/usr/bin/env ruby#${mastodon-gems.wrappedRuby}/bin/ruby#g' bin/$b
+      fi
+    done
+    for b in $(ls ${mastodon-gems}/bin/)
+    do
+      if [ ! -f bin/$b ]; then
+        ln -s ${mastodon-gems}/bin/$b bin/$b
+      fi
+    done
+
+    rm -rf log
+    ln -s /var/log/mastodon log
+    ln -s /tmp tmp
+  '';
+  propagatedBuildInputs = [ imagemagick ffmpeg file ];
```

```suggestion
  propagatedBuildInputs = [ imagemagick ffmpeg file mastodon-gems.wrappedRuby ];
```
This makes the wrapped ruby executable available to `patchShebangs`.